### ISSUE TITLE
[REVIEW]: Extracting, Computing and Exploring the Parameters of Statistical Models using R

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2445]
**Submitting author:** @strengejacke (<a href="http://orcid.org/0000-0002-8895-3206">Daniel Lüdecke</a>)
**Repository:** <a href="https://github.com/easystats/parameters" target ="_blank">https://github.com/easystats/parameters</a>
**Version:** 0.8.3
**Editor:** @sbenthall
**Reviewers:** @cmaimone, @Ohlsen
**Archive:** <a href="https://doi.org/10.5281/zenodo.4004846" target="_blank">10.5281/zenodo.4004846</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/158e1c7d5be57a0a70f137ca6cbf6912/status.svg)](https://joss.theoj.org/papers/158e1c7d5be57a0a70f137ca6cbf6912)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/158e1c7d5be57a0a70f137ca6cbf6912"><img src="https://joss.theoj.org/papers/158e1c7d5be57a0a70f137ca6cbf6912/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/158e1c7d5be57a0a70f137ca6cbf6912/status.svg)](https://joss.theoj.org/papers/158e1c7d5be57a0a70f137ca6cbf6912)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/easystats/parameters" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@cmaimone, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @sbenthall know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/easystats/parameters">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@strengejacke) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/easystats/parameters">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@strengejacke) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @cmaimone it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.21105/joss.01541 is OK
- 10.21105/joss.01412 is OK
- 10.18637/jss.v016.i09 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02445/joss.02445/10.21105.joss.02445.pdf)

**Comment by strengejacke** [comment]
@sbenthall Do you know why @Ohlsen has been unassigned as reviewer?

**Comment by sbenthall** [comment]
@strengejacke I don't. Let me ask ... @juanjo any idea what happened here?

**Comment by strengejacke** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02445/joss.02445/10.21105.joss.02445.pdf)

**Comment by strengejacke** [comment]
@cmaimone Dear Christina, thank you for your reviews regarding our package (website) and the paper. We have addressed your issues and included a response to your requests in the related GitHub issues (https://github.com/easystats/parameters/issues/274, https://github.com/easystats/parameters/issues/276 and https://github.com/easystats/parameters/issues/277).

We hope that our revisions adequately address your proposed changes. Please let us know if there are any further things you like us to revise.

**Comment by strengejacke** [comment]
@whedon generate pdf

**Comment by strengejacke** [comment]
We did some language revisions to the paper as well now.

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02445/joss.02445/10.21105.joss.02445.pdf)

**Comment by strengejacke** [comment]
@whedon generate pdf

**Comment by strengejacke** [comment]
We created Zenodo-repositories to assign our packages with a DOI, so all references should have a DOI now (where possible).

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02445/joss.02445/10.21105.joss.02445.pdf)

**Comment by cmaimone** [comment]
My review is complete.  Recommend accept.

**Comment by strengejacke** [comment]
@whedon generate pdf

**Comment by strengejacke** [comment]
@sbenthall I have made all corrections to the paper and also checked correct spelling/upper-lower-case in references - all should be fine now. What would be the next step?

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02445/joss.02445/10.21105.joss.02445.pdf)

**Comment by sbenthall** [comment]
@whedon assign Ohlsen as reviewer

**Comment by whedon** [comment]
OK, Ohlsen is now a reviewer

**Comment by sbenthall** [comment]
Hmm. That didn't work. I'm trying to reassign @Ohlsen . Sorry, I'm new here.

**Comment by sbenthall** [comment]
@whedon add @cmaimone as reviewer

**Comment by whedon** [comment]
OK, @cmaimone is now a reviewer

**Comment by sbenthall** [comment]
@whedon add @Ohlsen as reviewer

**Comment by whedon** [comment]
OK, @Ohlsen is now a reviewer

**Comment by sbenthall** [comment]
Ok, I think maybe @Ohlsen did something manually that unassigned them.
But now I've reassigned them. I believe they need to complete their review in addition to @cmaimone in order to proceed.

@xuanxu please let me know if that doesn't look right for some reason.

**Comment by xuanxu** [comment]
@sbenthall I've added manually the review checklist for @Ohlsen in the body of the issue. Everything looks ok now.

**Comment by strengejacke** [comment]
Thanks!

**Comment by strengejacke** [comment]
@sbenthall @Ohlsen After revising the paper and package based on the first review, CRAN requested an update from the _parameters_ package. Thus, I needed to increase the version number to indicate a "stable" release, and the GitHub-version will be 0.8.2.x the next days, and no longer 0.8.1.x - so don't be confused, @Ohlsen , when you start your review.

**Comment by sbenthall** [comment]
This is still waiting on @Ohlsen for review

**Comment by Ohlsen** [comment]
Hi, @sbenthall, apparently i missed some notification and am also not able to edit the checklist. Could you start the invitation again? Sorry for the trouble my self-assignment caused. I´m new to this as well ...

**Comment by Ohlsen** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by Ohlsen** [comment]
I´m still not able to edit the checklist. Perhaps it might work if i´m re-invited and not only re-assigned?

**Comment by strengejacke** [comment]
@xuanxu Do you have a guess, why @Ohlsen cannot edit the review-checkboxes?

**Comment by xuanxu** [comment]
Maybe the invitation expired? Let's try to re-invite

**Comment by xuanxu** [comment]
@whedon re-invite @Ohlsen as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@ohlsen please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by Ohlsen** [comment]
I finished my review, too. I had some feature requests focusing in particular on the packages functionality towards lavaan-models, which are already solved . I recommend accept as well.

**Comment by strengejacke** [comment]
Thanks, @Ohlsen, for your suggestions. Regarding your issues:

1) We revised the output for structural equation models, so `model_parameters()` now also include self-defined parameters (https://github.com/easystats/parameters/issues/290).
2) Since the vignettes on `model_parameters()` for the _lavaan_ package only focussed on CFA, we also added a small example for SEM as well.
3) We fixed some typos found in the paper.
4) To improve support for structural equation models, we revised `model_parameters()` so it can include different types of standardized solutions (https://github.com/easystats/parameters/issues/291).

We hope we adequately addressed all issues!

**Comment by sbenthall** [comment]
Thank you, reviewers.

@strengejacke , as a next step, would you please : 
* issue a new tagged release of the software (if changed),
* archive it (on Zenodo, figshare, or other).
* post the version number and archive DOI in this REVIEW issue.

**Comment by strengejacke** [comment]
Thank you @sbenthall 

- new release with latest changes from the revisions of this review is tagged: https://github.com/easystats/parameters/releases/tag/0.8.3
- archive for latest release on Zenodo created: https://zenodo.org/record/4004846#.X0gIBMgzaUk
- Version number: 0.8.3; DOI: 10.5281/zenodo.4004846

**Comment by strengejacke** [comment]
@sbenthall I have finished all requested steps, I think. Can you confirm that no further action from my side is required at the moment?

**Comment by sbenthall** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02445/joss.02445/10.21105.joss.02445.pdf)

**Comment by sbenthall** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/joss.01541 is OK
- 10.21105/joss.01412 is OK
- 10.18637/jss.v016.i09 is OK
- 10.1177/2515245918770963 is OK
- 10.5281/zenodo.3952174 is OK
- 10.5281/zenodo.3952153 is OK
- 10.5281/zenodo.3952214 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by sbenthall** [comment]
@whedon set 10.5281/zenodo.4004846 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4004846" target="_blank">10.5281/zenodo.4004846</a> is the archive.

**Comment by sbenthall** [comment]
@whedon set 0.8.3 as version

**Comment by whedon** [comment]
OK. 0.8.3 is the version.

**Comment by sbenthall** [comment]
This paper is ready to be accepted.

Before I do so finally, may I ask you @strengejacke to please check the proof one last time?

This is for your full satisfaction with the article, as making changes after the acceptance is more difficult.

**Comment by strengejacke** [comment]
Thank you, @sbenthall. I carefully proof-read the paper. Everything is fine for me, I'm ok with publication.

**Comment by danielskatz** [comment]
@sbenthall  - it looks like this is ready for you to ```@whedon accept``` and then I can take it, do a final proofread, and then publish it

**Comment by sbenthall** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/joss.01541 is OK
- 10.21105/joss.01412 is OK
- 10.18637/jss.v016.i09 is OK
- 10.1177/2515245918770963 is OK
- 10.5281/zenodo.3952174 is OK
- 10.5281/zenodo.3952153 is OK
- 10.5281/zenodo.3952214 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1721

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1721, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by sbenthall** [comment]
Your submission is accepted, @strengejacke 

Thank you for your article!

**Comment by strengejacke** [comment]
Thanks for editing this publication!

**Comment by danielskatz** [comment]
> Your submission is accepted, @strengejacke
> 
> Thank you for your article!

@sbenthall - FYI, this is not accepted, it's "recommended for acceptance".  See the label...

Accepted happens when it is published, which I hope will happen soon :)

**Comment by danielskatz** [comment]
@strengejacke  - please update the metadata in the Zenodo archive so that the title matches the paper title.

**Comment by danielskatz** [comment]
Also, for the Saefken reference, please add the URL to the arxiv abstract

**Comment by danielskatz** [comment]
Once these are done, I will finalize the acceptance and publication

**Comment by strengejacke** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02445/joss.02445/10.21105.joss.02445.pdf)

**Comment by strengejacke** [comment]
I revised the title of the Zenodo archive and added the URL. Generating PDF again to check if URL is fine...

**Comment by strengejacke** [comment]
Yes, URL is ok!

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1303742793186185216) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1722
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02445](https://doi.org/10.21105/joss.02445)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to @strengejacke (Daniel Lüdecke) and co-authors!!

Thanks to @cmaimone & @Ohlsen for reviewing, and @sbenthall for editing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02445/status.svg)](https://doi.org/10.21105/joss.02445)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02445">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02445/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02445/status.svg
   :target: https://doi.org/10.21105/joss.02445
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02445/status.svg)](https://doi.org/10.21105/joss.02445)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
